src/components/Office/ShipmentDetails/ShipmentDetailsMain.jsx

```diff
@@ -77,6 +78,27 @@ const ShipmentDetailsMain = ({
 
   const pendingSITExtension = sitExtensions?.find((se) => se.status === SIT_EXTENSION_STATUS.PENDING);
 
+  /**
+   * @description This variable is used to store the existence of service items
+   * codes that contain SIT and are ar part of the constants
+   * SERVICE_ITEM_CODES. It is false by default unless there are MTO service
+   * items to iterate over.
+   *
+   * When there are MTO service items present, we filter through all the
+   * service items and check the length of the new array against whether or not
+   * there are SIT suffixes in the service items codes and whether it matches
+   * the service items codes we have defined as constants.
+   */
+  let serviceItemsContainSIT = false;
+  if (mtoServiceItems) {
+    serviceItemsContainSIT = !!(
+      mtoServiceItems.filter((serviceItem) => {
+        const rgSIT = /SIT$/;
+        return rgSIT.test(serviceItem.reServiceCode) && !!SERVICE_ITEM_CODES[serviceItem.reServiceCode];
+      }).length > 0
+    );
+  }
```

Comment:
suggestion (blocking): I have a simplified suggestion below. Please take a look.

```suggestion
// This can be added in src/constants/serviceItems.js
// We know there are only six SIT service items
// Make a sub-list
const SIT_SERVICE_ITEM_CODES = {
  DDASIT: 'DDASIT',
  DDDSIT: 'DDDSIT',
  DDFSIT: 'DDFSIT',
  DOASIT: 'DOASIT',
  DOFSIT: 'DOFSIT',
  DOPSIT: 'DOPSIT',
};

const serviceItemsContainSIT = mtoServiceItems?.find(
    (serviceItem) => SIT_SERVICE_ITEM_CODES[serviceItem.reServiceCode],
  );
```

Reply:
Thanks for pointing out the broken tests. I'll fix that. I don't agree with your solution, but I can include a sub-list of SIT service item codes. 

My disagreement here stems from the the 3 lines of code at the bottom being much less readable but also changing the return value here from a boolean to a service item with SIT. I'll try to find a balance between the two ideas here though.

Reply:
I do think the suggested code is more readable, at least to me. Gonna tag @michaelvaldes-truss for a third opinion to see what they think.

Reply:
I refactored this to be more like what you wrote in the end here @donaldthai. My solutions still had us going through the entire Array, which is something I liked about your solution here. 

I re-worded the documentation around this to break apart the 3 lines you suggested a bit better for me. My original intention here was to actual have booleans for the true/false of whether it should display, but this is JS and it doesn't care. Your solution where the values are truthy (an object 🆚 undefined) is the better approach.

Thanks for the feedback.